tests/unit_tests/pandas_postprocessing/test_contribution.py

```diff
@@ -74,7 +74,7 @@ def test_contribution():
         rename_columns=["pct_a"],
     )
     assert processed_df.columns.tolist() == ["a", "b", "c", "pct_a"]
-    assert_array_equal(processed_df["a"].tolist(), [1, 3, nan])
-    assert_array_equal(processed_df["b"].tolist(), [1, 9, nan])
-    assert_array_equal(processed_df["c"].tolist(), [nan, nan, nan])
+    assert_array_equal(processed_df["a"].tolist(), [1, 3, 0])
+    assert_array_equal(processed_df["b"].tolist(), [1, 9, 0])
+    assert_array_equal(processed_df["c"].tolist(), [0, 0, 0])
```

Comment:
I don't think we want this change... a `nan` is very different from a 0.

Reply:
@zhaoyongjie how do you feel about this change? It seems weird that `nan` would now be `0` here. While it may be ok in this case, it would be important to understand the mechanism behind this change, as it might be introducing another regression that we don't have test coverage for right now..

Reply:
@villebro in this case this is the result of the `contribution` post processing function, so I think it makes sense — if something is not present it's contribution is zero. But I agree, we definitely don't want nans everywhere to become zeros... it would be nice to have more coverage on tests that manipulate data to ensure we're not returning wrong results (the worst kind of bug!).

Reply:
@betodealmeida I agree completely, for this case it does make sense IMO. If needed I can put aside some time this week for stepping through the contribution code to see where the difference is happening so we can pinpoint what's changed. I probably didn't say it here yet, but this is a great PR and helps keep Superset up to date, so I really appreciate the work done here by @EugeneTorap 🙂👍

Reply:
I looked more closely at this, and I think this needs to be fixed. The test is making sure that columns `a`, `b` and `c` are unchanged, and that the contribution of `a` should be placed in a new column `pct_a`. Therefore this would in fact count as a regression, as the original values should not be mutated. So I would make sure the test does the following:
- these three assertions are unchanged
- add an additional assertion to make sure `pct_a` is in fact the correct contribution. I expect it should be `[0.25, 0.75, 0]`

@EugeneTorap let me know if you need help with this.

Reply:
Hi @EugeneTorap , I am tracking this test case, the `nan` changed to `0` from [here](https://github.com/apache/superset/blob/4f1996dba8e35ee958048b726750247ec8e518aa/superset/utils/pandas_postprocessing/contribution.py#L51-L52). let's look over what's changes from `select_dtypes` in the upgrade.

Reply:
@zhaoyongjie it looks like `select_dtypes` used to return a copy, but now returns a reference to the original series, making the changes happen in place. I suspect it's caused by this: https://github.com/pandas-dev/pandas/pull/42611

Reply:
I'm concerned that you are blocking the change which is fixing the current broken installation :(

Reply:
Yes, I need the help. Anyone who has write access can add the fix
![image](https://user-images.githubusercontent.com/29536522/184587714-6bc8ab52-90a6-4aec-8466-9d7d9591379c.png)

Reply:
> @ghrom to be clear, without addressing the breaking change caused by upgrading to Pandas 1.4 we would be introducing a regression. I have asked @EugeneTorap if he needs help addressing the issue and I am happy to help if needed.

Please introduce yourself to the concept of the greater vs. lesser evil. The current state of the pip repository is BROKEN. You are worried about your tests?

Reply:
@ghrom Python 3.10 is not currently officially supported by Superset; please refer to the currently supported versions: https://github.com/apache/superset/blob/394d62ee51b6feb6e23c2b24505ee469460db146/setup.py#L183-L186
The project is happy to help add support for Python 3.10, but we simply can't merge this PR before ensuring that the regressions that it introduces have been fixed. Again, I'm very happy that this test exists, as otherwise we probably would have already merged this PR, introducing a regression.

In the meantime, use a supported version of Python when running Superset (`pyenv` and `virtualenv` are great tools for this).

Reply:
Your buggy tests are failing the project. Good luck with your 'great tools' perpetuating the dll hell.

Reply:
> The test is making sure that columns `a`, `b` and `c` are unchanged, and that the contribution of `a` should be placed in a new column `pct_a`.

Ah, I missed that, I thought those were contributions. Good catch!

Reply:
FYI I opened a bug report detailing the breaking change: https://github.com/pandas-dev/pandas/issues/48090

Reply:
@villebro Can we rewrite our logic to not depends it is zero or nan?

Reply:
> @villebro Can we rewrite our logic to not depends it is zero or nan?

@EugeneTorap sure we can, I can probably do it tomorrow. But I wanted to make the Pandas project aware of the regression first to also know what the plan is going forward; is this an intended breaking change or will it be fixed in a forthcoming patch release.

Reply:
FYI I've pushed the fix that makes the test changes unnecessary and I'm following up with @EugeneTorap to resolve a remaining issue